src/test/java/com/rocketden/main/service/SubmitServiceTests.java

```diff
@@ -43,6 +61,43 @@
     @InjectMocks
     private SubmitService submitService;
 
+    @Test
+    public void runCodeSuccess() {
+        Room room = new Room();
+        room.setRoomId(ROOM_ID);
+        User user = new User();
+        user.setNickname(NICKNAME);
+        user.setUserId(USER_ID);
+        room.addUser(user);
+
+        Game game = GameMapper.fromRoom(room);
+
+        List<Problem> problems = new ArrayList<>();
+        Problem problem = new Problem();
+        problem.setName(NAME);
+
+        ProblemTestCase testCase = new ProblemTestCase();
+        testCase.setInput(INPUT);
+        testCase.setOutput(OUTPUT);
+        problem.addTestCase(testCase);
+        problems.add(problem);
+        game.setProblems(problems);
+
+        SubmissionRequest request = new SubmissionRequest();
+        request.setLanguage(LANGUAGE);
+        request.setCode(CODE);
+        request.setInput(INPUT);
+        request.setInitiator(UserMapper.toDto(user));
+
+        SubmissionDto submissionDto = submitService.runCode(game, request);
+
+        assertEquals(CODE, submissionDto.getCode());
+        assertEquals(LANGUAGE, submissionDto.getLanguage());
+        assertEquals(submissionDto.getNumCorrect(), submissionDto.getNumTestCases());
+        assertNotNull(submissionDto.getRuntime());
+        assertFalse(game.getAllSolved());
```

Comment:
Now that we've added the results object to the Submission classes, could we add checks for those fields? (might require updating the getDummySubmission method to set those fields too)

Reply:
Yes, sounds good - done!